Library/Homebrew/rubocops/deprecate.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits deprecate!
+      class Deprecate < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          deprecate_node = find_node_method_by_name(body_node, :deprecate!)
+
+          return if deprecate_node.nil? || deprecate_node.children.length < 3
+
+          date_string = find_strings(deprecate_node).first
+
+          return if regex_match_group(date_string, /^\d{4}-\d{2}-\d{2}$/)
```

Comment:
I would suggest using `Date.iso8601`, something like

```ruby
          begin
            Date.iso8601(date_string)
          rescue ArgumentError
            offending_node...
            problem ...
          end
```

And probably problem message could include `Date.parse(date_string).iso8601` value as an expected result (we could add an autocorrect later)

Reply:
Ahh, didn't realize those were options. I'll play around with it!

Reply:
Even nicer would be to use `Date.parse` in the an autocorrect and convert it automatically to the right format!